actions/publish-github-pages/entrypoint.py

```diff
@@ -0,0 +1,246 @@
+                    # create _index.md file for the folder
+                    index_string = f"---\ntitle: \"{markdown_basename.capitalize()}\"\ndescription: \"\"\nicon: \"fa fa-brain\"\nweight: 1\n---"
+                    index_path = os.path.join(STAGING_DIR, "content", markdown_basename, "_index.md")
+                    with open(index_path, "w")as index_file:
+                        index_file.write(index_string)
+            # Otherwise, we're gonna preserve the existing file heirarchy.
+            else:
+                markdown_files = [potential_md_file for potential_md_file in files if ".md" in potential_md_file]
+                staging_root = root.replace(GITHUB_PAGES_PULL_PATH, "").lstrip('/')
+                weight = 1
+                for markdown_file in markdown_files:
+                    origin_path = os.path.join(root, markdown_file)
+                    destination_path = os.path.join(STAGING_DIR, "content", staging_root, markdown_file)
+                    insert_frontmatter(origin_path, weight=weight)
+                    os.renames( origin_path, destination_path )
+                    if DEBUG:
+                        print(f'origin: {origin_path}, destination: {destination_path}')
+                        #print(f'origin dir contents: {os.listdir(origin_path.rsplit("/",1)[0])}')
+                        #with open(destination_path, "r") as md_file:
+                        #    md_file_contents = md_file.read()
+                        #    print(md_file_contents)
+                    #weight = weight + 1
+
+    # Build Docker image needed to build the Hugo site
+    docker_build_command = f'cd {STAGING_DIR}; docker build -t cloudposse/docs .'
+    subprocess.run(docker_build_command, shell=True, check=True)
+
+    # publish the Hugo-generated HTML to $GITHUB_PAGES_PUSH_PATH
+    make_command = f'cd {STAGING_DIR}; make release; make real-clean hugo/build'
+    if DEBUG:
+        print(make_command)
+    subprocess.run(make_command, shell=True, check=True)
+    print(f"HUGO_PUBLISH_DIR: {HUGO_PUBLISH_DIR}, GITHUB_PAGES_PUSH_PATH: {GITHUB_PAGES_PUSH_PATH}")
+    copytree( os.path.join(STAGING_DIR, HUGO_PUBLISH_DIR), GITHUB_PAGES_PUSH_PATH, dirs_exist_ok=True )
+    print(f"Copied files from {os.path.join(STAGING_DIR, HUGO_PUBLISH_DIR)} to {GITHUB_PAGES_PUSH_PATH}")
+    print(f"Contents of {os.path.join(STAGING_DIR, HUGO_PUBLISH_DIR)}: {os.listdir(os.path.join(STAGING_DIR, HUGO_PUBLISH_DIR))}")
+    print(f"Contents of {GITHUB_PAGES_PUSH_PATH}: {os.listdir(GITHUB_PAGES_PUSH_PATH)}")
+
+def insert_frontmatter(file_path, weight=1):
+    # check for frontmatter
+    frontmatter_flag = True
+    with open(file_path, "r") as markdown_file:
+        line = markdown_file.readline()
+        if not re.match("^---", line):
+            frontmatter_flag = False
+    # if it's not found, create it by finding the first commented line and formatting it as frontmatter
```

Comment:
But it's not a commented line ;) It's the way headers are expressed in markdown.
```suggestion
    # if it's not found, create it by finding the first commented line and formatting it as frontmatter
```